flink/src/test/java/org/apache/iceberg/flink/TestHelpers.java

```diff
@@ -62,9 +61,8 @@ private TestHelpers() {
   }
 
   public static RowData copyRowData(RowData from, RowType rowType) {
-    ExecutionConfig config = new ExecutionConfig();
     TypeSerializer[] fieldSerializers = rowType.getChildren().stream()
-        .map((LogicalType type) -> InternalSerializers.create(type, config))
+        .map((LogicalType type) -> InternalSerializers.create(type))
```

Comment:
I'd prefer to create our iceberg's `RowDataCloner` (which is similar to the [RowDataConvert](https://github.com/apache/iceberg/blob/78e80d2a35c4c93e68776c31c24cc8ccb06fed4b/flink/src/test/java/org/apache/iceberg/flink/RowDataConverter.java#L49) ),  so that we could get ride of this flink internal `InternalSerializers`. 

Relying on flink internal interfaces that do not promise compatibility should be avoided as much as possible in subsequent development codes.

Reply:
It will be great if we can achieve the goal. Looking at the behavior of clone, I am wondering if we will need to reinvent a lot of specific serializers (like `BooleanSerializer`) that `InternalSerializers` eventually depends on. Those type serializers are also internal just like `InternalSerializers`.

Ideally, it will be great if `RowData` interface provides a public `clone` API. Actually, in the FLIP-27 source, we need need to clone `RowData` since RowDataIterator returns a reused `RowData` object.

Reply:
Here we will use this to copy a `RowData` into a newly created `GenericRowData`, that means we don't have to use the complex `TypeSerializer` ( that is used for copying binary data for `BinaryRowData`).   I think it will be good to remove the internal API if possible,  so I'd recommend to use the  similar `RowDataConverter` way to copy a totally new java objects for each fields. 

BTW, I've discussed with flink sql team (@wuchong),   the future apache flink will provides RowData copy utility to acomplish this RowData clone..

Reply:
Of course, we could open separate PR to address the `InternalSerializers` issue if you think it's necessary.

Reply:
Created issue-2184 to track this issue. probably a separate PR is better.